libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -881,9 +881,14 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 					// We do not visit the expression here on purpose, because in the case of an
 					// internal library function call, this would push the library address forcing
 					// us to link against it although we actually do not need it.
-					auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration);
-					solAssert(!!function, "Function not found in member access");
-					m_context << m_context.functionEntryLabel(*function).pushTag();
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						m_context << m_context.functionEntryLabel(*function).pushTag();
+					else if (dynamic_cast<EventDefinition const*>(_memberAccess.annotation().referencedDeclaration))
```

Comment:
This more or less mixes the location of the function type and the polymorphic type of the referenced declaration. As we are in the code generator here, it would perhaps be better to only select on basis of `funType->location()`, i.e. have one case (this block here) for internal, one for external and one for function.

This block of code might also be from the time where we only had "internal" and "external", please check if the "else" cas is not too permissive (by running the tests with an assertion in the else case).